File: AndroidDatabase/DataAdapterTest/app/src/main/java/com/haoyudu/dataadaptertest/model/MyViewModel.java
package com.haoyudu.dataadaptertest.model;

import android.app.Application;
import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import com.haoyudu.dataadaptertest.model.repository.ModelRepository;
import com.haoyudu.dataadaptertest.word.Word;

import java.util.List;

public class MyViewModel extends AndroidViewModel{

    private final ModelRepository modelRepository;

    public MyViewModel(@NonNull Application application) {
        super(application);
        modelRepository=new ModelRepository(application);
    }

    public void insertWords(Word...words){
        modelRepository.insertWords(words);
    }

    public Word queryWord(String wordName){
        return modelRepository.queryWord(wordName);
    }

    public void deleteWord(Word...words){
        modelRepository.deleteWord(words);
    }

    public void clearWords(){
        modelRepository.clearWords();
    }

    public void deleteWordLatest(){
        modelRepository.deleteWordLatest();
    }

    public boolean wordIsHave(String wordName){
        return modelRepository.wordIsHave(wordName);
    }


    public LiveData<List<Word>> getAllWords() {
        return modelRepository.getAllWords();
    }
}

File: KotlinTest/KotlinDatabase/app/src/main/java/com/haoyudu/kotlindatabase/MyViewModel.kt
package com.haoyudu.kotlindatabase

import android.app.Application
import android.os.AsyncTask
import android.view.View
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.room.Room

class MyViewModel(application: Application) : AndroidViewModel(application) {
    private var viewModelRepository:ViewModelRepository= ViewModelRepository(application)

    fun insert(word:Word){
        viewModelRepository.insert(word)
    }

    fun queryAllWords(){
        viewModelRepository.queryAllWords()
    }

    fun clearAllWords(){
        viewModelRepository.clearAllWords()
    }

    fun getAllWords():LiveData<List<Word>>{
        return viewModelRepository.getAllWords()
    }

}